kuma/wiki/tests/test_views.py

```diff
@@ -2281,48 +2282,180 @@ def test_quick_review(self):
+            doc = Document.objects.get(slug=data['slug'])
+
+            # Edit #1 starts...
+            resp = self.client.get(reverse('wiki.edit',
+                                           args=[doc.slug]))
+            page = pq(resp.content)
+            rev_id1 = page.find('input[name="current_rev"]').attr('value')
+
+            # Edit #2 starts...
+            resp = self.client.get(reverse('wiki.edit',
+                                           args=[doc.slug]))
+            page = pq(resp.content)
+            rev_id2 = page.find('input[name="current_rev"]').attr('value')
+
+            # Edit #2 submits successfully
+            data.update({
+                'form-type': 'rev',
+                'content': 'This edit got there first',
+                'current_rev': rev_id2
+            })
+            if is_ajax:
+                resp = self.client.post(
+                    reverse('wiki.edit', args=[doc.slug]),
+                    data, HTTP_X_REQUESTED_WITH='XMLHttpRequest'
+                )
+                eq_(False, json.loads(resp.content)['error'])
+            else:
+                resp = self.client.post(reverse('wiki.edit',
+                                                args=[doc.slug]), data)
+                eq_(302, resp.status_code)
+
+            # Edit #1 submits, but receives a mid-aired notification
+            data.update({
+                'form-type': 'rev',
+                'content': 'This edit gets mid-aired',
+                'current_rev': rev_id1
+            })
+
+            if is_ajax:
+                resp = self.client.post(
+                    reverse('wiki.edit', args=[doc.slug]),
+                    data,
+                    HTTP_X_REQUESTED_WITH='XMLHttpRequest'
+                )
+            else:
+                resp = self.client.post(reverse('wiki.edit',
+                                                args=[doc.slug]), data)
+
+            history_url = reverse(
+                'wiki.document_revisions',
+                kwargs={'document_path': doc.slug}, locale=doc.locale)
+            midair_collission_error = (unicode(MIDAIR_COLLISION) % {'url': history_url}).encode('utf-8')
+
+            if is_ajax:
+                location_of_error = json.loads(resp.content)['error_message']
+            else:
+                location_of_error = resp.content
+            ok_(midair_collission_error in location_of_error,
+                "Midair collision message should appear")
+
+        midair_collision_test(is_ajax=False)
+        midair_collision_test(is_ajax=True)
+
+    @override_config(AKISMET_KEY='dashboard')
+    @requests_mock.mock()
+    @mock.patch('kuma.spam.akismet.Akismet.check_comment')
+    def test_edit_spam_ajax(self, mock_requests, mock_akismet_method):
+        """Tests attempted spam edits that occur on Ajax POSTs."""
+        # Enable Akismet
+        Flag.objects.create(name=SPAM_SUBMISSIONS_FLAG, everyone=True)
```

Comment:
Use the [@override_flag decorator](http://waffle.readthedocs.io/en/v0.11/testing/automated.html#unit-tests) instead of creating flags in tests.